File: app/src/main/java/com/sortedqueue/programmercreek/adapter/LanguageRecyclerAdapter.java
package com.sortedqueue.programmercreek.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.sortedqueue.programmercreek.R;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by Alok Omkar on 2017-04-16.
 */

public class LanguageRecyclerAdapter extends RecyclerView.Adapter<LanguageRecyclerAdapter.ViewHolder> {

    private ArrayList<String> languages;
    private CustomProgramRecyclerViewAdapter.AdapterClickListner adapterClickListner;
    private String selectedLanguage = "";

    public LanguageRecyclerAdapter(ArrayList<String> languages,
                                   CustomProgramRecyclerViewAdapter.AdapterClickListner adapterClickListner) {
        this.languages = languages;
        this.adapterClickListner = adapterClickListner;
    }

    public String getSelectedLanguage() {
        return selectedLanguage;
    }

    public void setSelectedLanguage(String selectedLanguage) {
        this.selectedLanguage = selectedLanguage;
        notifyDataSetChanged();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_tag, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        String language = languages.get(position);
        holder.tagTextView.setText(language);
        holder.tagLayout.setSelected( selectedLanguage.equals(language));
    }

    @Override
    public int getItemCount() {
        return languages.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        @BindView(R.id.tagTextView)
        TextView tagTextView;
        @BindView(R.id.tagLayout)
        LinearLayout tagLayout;

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            int position = getAdapterPosition();
            if( position != RecyclerView.NO_POSITION ) {
                adapterClickListner.onItemClick(position);
            }
        }
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/adapter/LanguageRecyclerAdapter.kt
package com.sortedqueue.programmercreek.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView

import com.sortedqueue.programmercreek.R

import java.util.ArrayList

import butterknife.BindView
import butterknife.ButterKnife

/**
 * Created by Alok Omkar on 2017-04-16.
 */

class LanguageRecyclerAdapter(private val languages: ArrayList<String>,
                              private val adapterClickListner: CustomProgramRecyclerViewAdapter.AdapterClickListner) : RecyclerView.Adapter<LanguageRecyclerAdapter.ViewHolder>() {
    var selectedLanguage = ""
        set(selectedLanguage) {
            field = selectedLanguage
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_tag, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val language = languages[position]
        holder.tagTextView!!.text = language
        holder.tagLayout!!.isSelected = this.selectedLanguage == language
    }

    override fun getItemCount(): Int {
        return languages.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        @BindView(R.id.tagTextView)
        internal var tagTextView: TextView? = null
        @BindView(R.id.tagLayout)
        internal var tagLayout: LinearLayout? = null

        init {
            ButterKnife.bind(this, itemView)
            itemView.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            val position = adapterPosition
            if (position != RecyclerView.NO_POSITION) {
                adapterClickListner.onItemClick(position)
            }
        }
    }
}
